functions/go/set-gcp-resource-ids/main.go

```diff
@@ -75,39 +52,38 @@ func (p *SetGCPProject) GenerateProjectID(objects fn.KubeObjects) (string, error
 	return projectID, nil
 }
 
-func (p *SetGCPProject) Transform(objects fn.KubeObjects) error {
+
+func (p *SetGCPProject) Run(ctx *fn.Context, functionConfig *fn.KubeObject, objects fn.KubeObjects) {
 	projectID := p.ProjectID
 	if projectID == "" {
 		// TODO: Only if we need a project id (though there aren't many cases where we don't)
-		p, err := p.GenerateProjectID(objects)
+		var err error
+		projectID, err = p.GenerateProjectID(objects)
 		if err != nil {
-			return err
+			ctx.ResultErrAndDie(err.Error(), nil)
 		}
-		projectID = p
 	}
 
 	packageContext, err := kpt.FindPackageContext(objects)
 	if err != nil {
-		return err
+		ctx.ResultErrAndDie(err.Error(), nil)
 	}
-
 	for _, object := range objects {
 		if object.IsLocalConfig() {
 			continue
 		}
 		if kpt.IsResourceGroup(object) {
 			continue // Should ResourceGroup be marked as local config?
 		}
```

Comment:
Another mysql style can be `for _, object := range objects.WhereNot(fn. IsLocalConfig).WhereNot(kpt. IsResourceGroup)`
But I like the current format which is easier to read.